### PR TITLE
[docs] update duckdb reference to be deps-aware

### PR DESCRIPTION
**Pull request by jamiedemaria**
## Summary & Motivation
Updating the duckdb reference page to have more info about the resource and be more deps aware

### CODE REVIEW

**Comment by jamiedemaria** [comment]
Current dependencies on/for this PR:
* master
  * **PR #15518** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15518" target="_blank">Graphite</a> 
    * **PR #15531** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15531" target="_blank">Graphite</a> 
      * **PR #15589** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15589" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/15589?utm_source=stack-comment).

File: docs/content/integrations/duckdb/reference.mdx
Comment (jamiedemaria) [inline docs/content/integrations/duckdb/reference.mdx:56]: how do we feel about this section? there's nothing particularly novel about using the resource in an op vs in an asset, but maybe this would be helpful for the op users out there? maybe it would just add confusion about when to use assets vs ops?

Comment (erinkcochran87) [reply docs/content/integrations/duckdb/reference.mdx:56]: I have feelings about this, but I'm not totally sure what those feelings are. Not helpful, I know.

1. **On ops vs assets**: On one hand, I like having an ops-specific example for coverage purposes, but on the other... we're really focusing on assets as the best way to use Dagster. Having this be so high up on the page (and presented as a sorta-binary against assets) feels contradictory to that.
2. **The heading change:** This was the `Executing custom SQL queries` example originally, right? IMO having the example lead with that is really valuable - I know @tacastillo has mentioned that folks ask about how to execute SQL on support not-infrequently. Re-naming this and having the description of what it's doing buried in the explanation may be detrimental.

   Perhaps just adding a sentence before the example itself would be helpful - something like "In this example, we're using the DuckDB resource to execute a custom SQL query"

File: docs/content/integrations/duckdb/reference.mdx
Comment (erinkcochran87) [inline docs/content/integrations/duckdb/reference.mdx:80]: > However, if you find that these do not...

It feels sorta nitpicky, but `these` feels kinda vague here. What's it referring to? The examples? The I/O manager itself?

File: docs/content/integrations/duckdb/reference.mdx
Comment (erinkcochran87) [inline docs/content/integrations/duckdb/reference.mdx:113]: [1] No rule is needed here, since the following heading is an H3.

File: docs/content/integrations/duckdb/reference.mdx
Comment (erinkcochran87) [inline docs/content/integrations/duckdb/reference.mdx:292]: [1]

File: docs/content/integrations/duckdb/reference.mdx
Comment (erinkcochran87) [inline docs/content/integrations/duckdb/reference.mdx:390]: [1]

File: docs/content/integrations/duckdb/reference.mdx
Comment (erinkcochran87) [inline docs/content/integrations/duckdb/reference.mdx:565]: [1]

File: docs/content/integrations/duckdb/reference.mdx
Comment (jamiedemaria) [reply docs/content/integrations/duckdb/reference.mdx:113]: ty ty i keep forgetting this

File: docs/content/integrations/duckdb/reference.mdx
Comment (jamiedemaria) [reply docs/content/integrations/duckdb/reference.mdx:80]: i wrote it intending "these" to be the customization options, but it would be an equally valid statement to be referring to the I/O manager itself.

File: docs/content/integrations/duckdb/reference.mdx
Comment (jamiedemaria) [reply docs/content/integrations/duckdb/reference.mdx:56]: 1 - i might just be overindexing on a recent user convo where they were trying to do bigquery resource stuff in ops and got a little confused. an example might have helped them but idk if that's enough justification to add it in 

for 2 - that's really valuable insight and i think a good reason to change the heading back to how it was before

File: docs/content/integrations/duckdb/reference.mdx
Comment (erinkcochran87) [reply docs/content/integrations/duckdb/reference.mdx:80]: Gotcha! I think something like "However, if you find these options..." would be an improvement, as it's then clear what's being referenced

File: docs/content/integrations/duckdb/reference.mdx
Comment (jamiedemaria) [reply docs/content/integrations/duckdb/reference.mdx]: I went ahead and removed the ops example and put the heading back to what it was before

File: docs/content/integrations/duckdb/reference.mdx
Comment (erinkcochran87) [inline docs/content/integrations/duckdb/reference.mdx:13]: Could you remove this guy?

**Review by erinkcochran87 — APPROVED** [review]
One small thing leftover from the ops example, but otherwise this lgtm. I'm gonna stamp this to unblock you, but up to you if you wanna wait for Yuhan or Tim to give their feedback too

File: docs/content/integrations/duckdb/reference.mdx
Comment (tacastillo) [inline docs/content/integrations/duckdb/reference.mdx:45]: Do we need the cursor to run a query? I feel like I've been directly calling `.execute` from the connection, but it's been a minute since I tried.

**Review by tacastillo — COMMENTED** [review]
Smol comment!

File: docs/content/integrations/duckdb/reference.mdx
Comment (jamiedemaria) [reply docs/content/integrations/duckdb/reference.mdx:45]: good point - i'm actually not sure so i will test

Comment (jamiedemaria) [reply docs/content/integrations/duckdb/reference.mdx:45]: i'm getting majorly weird issues trying to run anything with duckdb - seems like the commands are just hanging forever with no output or errors... and it's happening with both `cursor()` and raw `execute()`. I think it's completely unrelated to the actual code snippet and is rather some strange issue with my duckdb setup. I'm going to timebox myself like 30 min to see if i can figure out what's up, and then go from there

Comment (tacastillo) [reply docs/content/integrations/duckdb/reference.mdx:45]: Oh, I've seen this before.

One of the reasons why is if you're making a new db file in a directory that doesn't exist, duckdb won't do the `mkdir -p` for you, and the Dagster step will hang forever.

Could that be it?

Comment (jamiedemaria) [reply docs/content/integrations/duckdb/reference.mdx:45]: oh maybe, i'll check that

Comment (jamiedemaria) [reply docs/content/integrations/duckdb/reference.mdx:45]: deleting the db and letting duckdb make it from scratch again did it. so something just got wonky somewhere. but the `cursor()` is not required! I'll remove it

Comment (tacastillo) [reply docs/content/integrations/duckdb/reference.mdx]: Oh, interesting. Do you think the duckdb versions can be different? Because DuckDB storage isn't compatible across patch versions, so I'm wondering if it's silently erroring or we're not catching the exception properly.

**Review by tacastillo — APPROVED** [review]
image

**Pull request merged**
